msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -2335,14 +2299,14 @@ public boolean isValidBrokerPackage(@NonNull final Context context,
     @Nullable
     public String getActiveBrokerPackageName(@NonNull final Context context)
             throws MsalException {
-        try {
-            return new BrokerValidator(context).getCurrentActiveBrokerPackageName();
-        } catch (final ClientException e){
-            if (NOT_VALID_BROKER_FOUND.equalsIgnoreCase(e.getErrorCode())) {
-                return null;
-            }
+        final BrokerData activeBroker = BrokerDiscoveryClientFactory.getInstance(context,
+                        AndroidPlatformComponentsFactory.createFromContext(context))
+                .getActiveBroker(false);
 
-            throw MsalExceptionAdapter.msalExceptionFromBaseException(e);
+        if (activeBroker != null) {
```

Comment:
The method is marked Nullable, so you could just get rid of the `if` statement and just return `activeBroker.getPackageName()`. (then can also get rid of line 2310)

Reply:
activeBroker itself could be null.